v2/pkg/manifest/oci-manifest.go

```diff
@@ -182,129 +173,160 @@ func untar(gzipStream io.Reader, path string, cfgDirName string) error {
 }
 
-// readOperatorConfig - reads the catalog.json and unmarshals it to DeclarativeConfig struct
-func readOperatorConfig(path string) ([]v1alpha3.DeclarativeConfig, error) {
-	var olm []v1alpha3.DeclarativeConfig
-	data, err := os.ReadFile(path + "/" + catalogJson)
-	if err != nil {
-		return []v1alpha3.DeclarativeConfig{}, err
+func newOperatorCatalog() OperatorCatalog {
+	operatorConfig := OperatorCatalog{
+		Packages:            make(map[string][]declcfg.Package),
+		Channels:            make(map[string][]declcfg.Channel),
+		ChannelEntries:      make(map[string]map[string]map[string]declcfg.ChannelEntry),
+		BundlesByPkgAndName: make(map[string]map[string]declcfg.Bundle),
 	}
-	tmp := strings.NewReplacer(" ", "").Replace(string(data))
-	updatedJson := "[" + strings.ReplaceAll(tmp, "}\n{", "},{") + "]"
-	err = json.Unmarshal([]byte(updatedJson), &olm)
-	if err != nil {
-		return []v1alpha3.DeclarativeConfig{}, err
-	}
-	return olm, nil
+
+	return operatorConfig
 }
 
-// getRelatedImages - get the packages' related images of a catalog.
-func getRelatedImages(log clog.PluggableLoggerInterface, operatorConfig []v1alpha3.DeclarativeConfig, iscOperator v1alpha2.IncludePackage, full bool) (map[string][]v1alpha3.RelatedImage, error) {
-	relatedImages := make(map[string][]v1alpha3.RelatedImage)
-	bundles := make(map[string]bool)
-	var defaultChannel string
+func parseOperatorCatalogByOperator(operatorName string, operatorConfig OperatorCatalog, operatorCatalog OperatorCatalog) OperatorCatalog {
+	operatorConfig.Packages[operatorName] = operatorCatalog.Packages[operatorName]
+	operatorConfig.Channels[operatorName] = operatorCatalog.Channels[operatorName]
+	operatorConfig.ChannelEntries[operatorName] = operatorCatalog.ChannelEntries[operatorName]
+	operatorConfig.BundlesByPkgAndName[operatorName] = operatorCatalog.BundlesByPkgAndName[operatorName]
+
+	return operatorConfig
+}
 
+func getRelatedImages(log clog.PluggableLoggerInterface, operatorName string, operatorConfig OperatorCatalog, iscOperator v1alpha2.IncludePackage, full bool) (map[string][]v1alpha3.RelatedImage, error) {
 	if isInvalidFiltering(iscOperator, full) {
 		return nil, fmt.Errorf("cannot use channels/full and min/max versions at the same time")
 	}
 
-	for i, obj := range operatorConfig {
-		switch {
-		case obj.Schema == "olm.channel":
-			if len(iscOperator.Channels) > 0 {
-				if found, idx := containsChannel(iscOperator.Channels, obj.Name); found {
-					log.Debug("found channel : %v", obj)
-					filteredBundles, err := filterBundles(obj.Entries, iscOperator.Channels[idx].IncludeBundle.MinVersion, iscOperator.Channels[idx].IncludeBundle.MaxVersion, full)
-					if err != nil {
-						log.Error(errorSemver, err)
-					}
-					for _, b := range filteredBundles {
-						bundles[b] = true
-					}
+	relatedImages := make(map[string][]v1alpha3.RelatedImage)
+	var filteredBundles []string
+	defaultChannel := operatorConfig.Packages[operatorName][0].DefaultChannel
```

Comment:
As the operator can have only one olm.package I will change it to have only one declarative config instead of an array of them. Thanks for the suggestion.